eslint.config.mjs

```diff
@@ -977,4 +979,143 @@ export default typescript.config([
       'no-restricted-imports': 'off',
     },
   },
+  {
+    name: 'plugin/boundaries',
+    plugins: {
+      boundaries,
+    },
+    settings: {
+      // order matters here in case of nested directories
+      'boundaries/elements': [
+        // --- specifics ---
+        {
+          type: 'devtoolbar',
+          pattern: 'sentry/components/devtoolbar',
+        },
+        {
+          type: 'core-button',
+          pattern: 'static/app/components/core/button',
+        },
+        {
+          type: 'core',
+          pattern: 'static/app/components/core',
+        },
+        // --- sentry ---
+        {
+          type: 'sentry-images',
+          pattern: 'static/images',
+        },
+        {
+          type: 'sentry-locale',
+          pattern: '(static/app/locale.tsx|src/sentry/locale/**/*.*)',
+          mode: 'full',
+        },
+        {
+          type: 'sentry-logos',
+          pattern: 'src/sentry/static/sentry/images/logos',
+        },
+        {
+          type: 'sentry-fonts',
+          pattern: 'static/fonts',
+        },
+        {
+          type: 'sentry-fixture',
+          pattern: 'tests/js/fixtures',
+        },
+        {
+          type: 'sentry',
+          pattern: 'static/app',
+        },
+        // --- getsentry ---
+        {
+          type: 'getsentry',
+          pattern: 'static/gsApp',
+        },
+        // --- admin ---
+        {
+          type: 'gsAdmin',
+          pattern: 'static/gsAdmin',
+        },
+        // --- tests ---
+        {
+          type: 'sentry-test',
+          pattern: 'tests/js/sentry-test',
+        },
+        {
+          type: 'getsentry-test',
+          pattern: 'tests/js/getsentry-test',
+        },
+        {
+          type: 'tests',
+          pattern: 'tests/js',
+        },
+        // --- configs ---
+        {
+          type: 'configs',
+          pattern: '(package.json|config/**/*.*|*.config.{mjs,js,ts})',
+          mode: 'full',
+        },
+        {
+          type: 'build-utils',
+          pattern: 'build-utils',
+        },
+        {
+          type: 'scripts',
+          pattern: 'scripts',
+        },
+      ],
+    },
+    rules: {
+      ...boundaries.configs.strict.rules,
+      'boundaries/no-ignored': 'off',
+      'boundaries/no-private': 'off',
+      'boundaries/element-types': [
+        'warn',
+        {
+          default: 'disallow',
+          message: '${file.type} is not allowed to import ${dependency.type}',
+          rules: [
+            {
+              from: ['sentry*'],
+              allow: ['core*', 'sentry*'],
```

Comment:
`from` and `allow` are `micromatch` patterns matched against the `type` defined in the above `settings`. I’ve “grouped” all `sentry` app related things as `”sentry-“` so that we can target them with `”sentry*”` here.

So this means: `sentry` can import from `sentry` (itself) and from everything `core`, and since the default is `disallow`, everything else is forbidden and will yield a warning.